Tasks/Maven/maventask.ts

```diff
@@ -105,18 +110,17 @@ if (specifiedJavaHome) {
     tl.setVariable('JAVA_HOME', specifiedJavaHome);
 }
 
-if (isCodeCoverageOpted) {
-    var summaryFile: string = null;
-    var reportDirectory: string = null;
-    var reportPOMFile: string = null;
-    var execFileJacoco: string = null;
-    var ccReportTask: string = null;
-    enableCodeCoverage();
-}
-else {
-    tl.debug("Option to enable code coverage was not selected and is being skipped.");
-}
+enableCodeCoverage()
+    .then(function (resp) {
+        ccReportTask = "jacoco:report";
+        tl.debug("Enabled code coverage successfully: " + ccReportTask);
+    }).catch(function (err) {
+        tl.warning("Failed to enable code coverage: " + err);
+    }).fin(function () {
+        execBuild();
```

Comment:
This separation is misleading - we don't enable the code coverage and then execute the build. We enable CC, do a mvn -version check, configure maven opts, configure SQ, configure code analysis etc... I suggest insteaad you create a ConfigureCodeCoverage() : Promise function and use that in the chain of promises below instead of creating "execBuild"